### ISSUE TITLE
[BUG] Rdp connection not happening

### ISSUE THREAD

**Issue by docop** [Zer0CoolX/guacamole-install-rhel#96]
From 3 different installs over cloud server, with version of yesterday 20 and today 21jan, installation from a fresh minimal 7.5 centos , install complete with no error present. Fully able to login and add user and connection. But when try to connect to rdp server, it don't connect at all. 
From an older script version : 
https://github.com/Zer0CoolX/guacamole-install-rhel/raw/2899e4afca05d572f59c661c229508bc16bf94dc/guac-install.sh
 it work fine with rdp, but ssl renewal fail. But while a bit old anyway, rdp do work. Same setting on both install.
I can't find the rdp version as mentionned in a closed/resolve ticket as ver 2.0 was not compatible or so.. But currently, no rdp connection are possible.

**Confirmation of Process**
1. You have read the [How to report issues](https://github.com/Zer0CoolX/guacamole-install-rhel/wiki/How-to-Report-Issues-(Bugs,-Feature-Request-and-Help)) wiki page? Y/N y
2. You have read the wiki and followed its guidelines including requirements, troubleshooting, etc.? Y/N y

**Details**
OS/Version: RHEL 7.5 min fresh , ping from the server ok, log to guac work too, create new user/connection work.
Guacamole Source: Stable/Git? stable / letsencrpy option, default all , 1 mariadb

thanks in advance

**Comment by Zer0CoolX** [comment]
You said you installed CentOS 7.5 and RHEL 7.5, after running the script have you been upgraded to 7.7 or higher? (cat /etc/redhat-release)

Make sure you are on the latest 7.x version before running the script. As per [Requirements](https://github.com/Zer0CoolX/guacamole-install-rhel/wiki/Requirements) I only test and support the latest 7.x versions.

Also make sure you are using the latest version of my script direct from this repo.

If on the latest 7.x and using the current version of the script and rdp still fails, I recommend reading over [Troubleshooting](https://github.com/Zer0CoolX/guacamole-install-rhel/wiki/Troubleshooting) AND I would recommend testing the simplest setup you can...IE: self signed cert, mariadb, as much default as possible and run.

I can confirm that on my current script and in CentOS 7.7 that RDP works for me (tested as I am writing this to be sure). I do not currently have the means to test with Lets Encrypt so if it works for you with self signed but breaks when you try with lets encrypt let me know.

Hope this helps.

**Comment by docop** [comment]
Hi So retry 2 other.. for sure, for me, now with : centos7.5 and then run update. then install your script and the one from yesterday and your latest give me error. But i found the last commit of the 15 nov:
https://github.com/Zer0CoolX/guacamole-install-rhel/raw/3d9a626e2d8668c5b11c28cbac236dfc1722a350/guac-install.sh
 this one do work and rdp pop up right away.
 ** for reference, on the latest, i've tried a vnc connection and it's ok, just rdp.
stable/ enter for the 3 def db name. / db pass as per example / 1 for sslletsencrypt / 1maria/ 1 no 2 factor top / and all default (just press enter) on the security and other.
thanks

**Comment by docop** [comment]
Also found out that on the 15nov version rdp do connect to linux xrdp, but get some issues with windows.. 
while the older one :+1: wget https://github.com/Zer0CoolX/guacamole-install-rhel/raw/2899e4afca05d572f59c661c229508bc16bf94dc/guac-install.sh
 do log straight away.

**Comment by docop** [comment]
Interesting.. so did another install with openssl, same setting option, no error present at install. And now rdp do work. If you choose Letsencrypt, rdp cannot connect.
 But if i install the last commit of 15nov, with letsencrypt option, it work.

**Comment by Zer0CoolX** [comment]
Sorry its a little hard to follow your comment. To be clear, are you saying it does work for you now using the latest version of the script?

If not, you again mention CentOS 7.5. Please be aware that I cannot help and do not support previous versions of 7.x with the exception of the latest, at this time 7.7. Please see my prior comment and follow the advice there.

If it is working now, feel free to close this issue unless you have other questions or concerns. Thanks.

**Comment by docop** [comment]
Hi Sorry, here a better wrapup. hehe
script 21jan - openssl = rdp work
script 21jan -LetsEncrypt = rdp do not work

script 15nov -letsEncrypt = rdp work

**Comment by Zer0CoolX** [comment]
Ok so it sounds like with the latest version of the script using Lets Encrypt, RDP has not worked for you?

As I mentioned, I currently have no way to test/troubleshoot with Lets Encrypt myself.

Are there any logs or info you can provide related to the issue?

Also, you still have not confirmed you are using the latest CentOS 7.7. This is critical...if your not using CentOS 7.7 then I can no longer help you.

**Comment by docop** [comment]
I do have a 7.5 based image and do the yum update right after install. So that should be the latest.. ?
 And looking for proper log.
thanks

**Comment by Zer0CoolX** [comment]
> I do have a 7.5 based image and do the yum update right after install. So that should be the latest.. ?

You need to confirm this. Run the command `cat /etc/redhat-release`. Do this before running my script and after updating CentOS 7.5, if its not 7.7 before running the script that's likely your issue.

If it is 7.7, then read my responses above and the wiki for information on where to get log information from and give me any relevant log info you can.

**Comment by vm1984** [comment]
Hi Guys.
I was installing guacamole yesterday with the latest version of script.
After the first try I was having problem with the rdp. I could not connect to the configured session.
/var/log/messages was telling me that:  Support for protocol "rdp" is not installed.
Then I have cloned the git repo for guacamole and ran the configure script manually.  I suppose that the latest version of guacamole is expecting freerdp version 2. 
After I have installed freerdp-2.0.0-1.rc4.el7.x86_64, freerdp-devel-2.0.0-1.rc4.el7.x86_64 and freerdp-libs-2.0.0-1.rc4.el7.x86_64 the configure script was fine - RDP ........... yes. 
So I did make and make install again manually, restarted the services and it starts working as expected. 
My version of centos is CentOS Linux release 7.7.1908
Lets encrypt was disabled.

**Comment by vm1984** [comment]
Maybe this will help.
When I ran ./configure
Guacamole server version 1.0.0 gave me - freerdp ............. yes
Guacamole  server version 1.1.0 gave me -  freerdp2 ............ yes
So its probably expecting freerdp version 2.0.0.
https://github.com/apache/guacamole-server/commit/28f7d3694e7be0bee4b4deb1cbf5e2a48b8f275b

**Comment by Zer0CoolX** [comment]
You said in your OP that you used stable, not git as the install source:

> Guacamole Source: Stable/Git? **stable** / letsencrpy option, default all , 1 mariadb

If using stable (1.0.0), it would have required freerdp1.x which should work. You can confirm RDP installed properly by looking in the install log, there will be an area while compiling Guacamole server that specifies what protocols are available/installed, basically like you mention you saw. It should say yes for RDP.

However, if you had the problem with installing from git and NOT stable, then this could explain it. Apache must have just updated 1.1.0 from git to use freerdp2 (been waiting for this).

So 2 things if you used git as source.

1. I do plan to update the script to account for changes like this periodically.
2. I do **not** provide support for using the git install source, only stable. Git is too unpredictable and not intended for use in production. The option is only in the script for testing purposes, mostly my own to test features before they release and update the script to get ready.

So please do the following to test and confirm if there is still an issue or not:

1. Use the latest version of the script.
2. Install CentOS 7.7. It doesn't matter if you do this from a 7.7 ISO/disc or from 7.5 and update, as long as after running `yum update` you reboot the computer AND confirm the version is 7.7 BEFORE running the script.
3. Try the following settings first: stable, self signed cert, nginx defaults, mariadb, no 2FA and no custom extension.
4. Enter the option to run the script to install.
5. For RDP, confirm that RDP works on the destination machine. IE: try to RDP to it from another machine using something like Microsofts RDP client.
6. Setup the connection in Guacamole, just the minimum IP (do not use hostname, use local IP) and the proper port, typically 3389.

IMPORTANT:
Before attempting to connect via RDP using Guacamole, open a terminal to the server and run the command `tail -f /var/log/messages`. After running this command, keep that terminal open and then try and connect via RDP in Guac. The terminal window will display messages about the connection. See this for details: https://github.com/Zer0CoolX/guacamole-install-rhel/issues/94#issuecomment-577700138

If it doesn't work let me know what it says. If it does work and you want Lets Encrypt you can repeat the process (clean install of CentOS) but using the Lets Encrypt option instead of the self signed option (keeping all else the same in step 3). and again verify if it works or not.

I hope this helps. As mentioned I see RDP working via the script and self signed cert and do not have the means to test Lets Encrypt right now (hoping to be able in a few weeks time). The keys to narrowing this down are checking the install log, checking other logs like /var/log/messages and ensuring the setup of things external to the script (like the requirements on the network for Lets Encrypt to get a cert) are done properly before hand.

**Comment by docop** [comment]
hi, so yes 7.7 version.
Here's the log from the tail, when i click connect rdp:

` Creating new client for protocol "rdp"
Jan 29 19:08:05 bc guacd: guacd[1244]: INFO:#011Creating new client for protocol "rdp"
Jan 29 19:08:05 bc guacd[1244]: Connection ID is "$e65a8898-bbaa-413e-9f10-c526a5f2a91d"
Jan 29 19:08:05 bc guacd: guacd[1244]: INFO:#011Connection ID is "$e65a8898-bbaa-413e-9f10-c526a5f2a91d"
Jan 29 19:08:05 bc guacd[9642]: Support for protocol "rdp" is not installed
Jan 29 19:08:05 bc guacd: guacd[9642]: WARNING:#011Support for protocol "rdp" is not installed
Jan 29 19:08:05 bc guacd[1244]: Connection "$e65a8898-bbaa-413e-9f10-c526a5f2a91d" removed.
Jan 29 19:08:05 bc guacd: guacd[1244]: INFO:#011Connection "$e65a8898-bbaa-413e-9f10-c526a5f2a91d" removed.

Jan 29 19:08:20 bc server: 19:08:20.653 [http-bio-8080-exec-42] ERROR o.a.g.w.GuacamoleWebSocketTunnelEndpoint - Creation of WebSocket tunnel to guacd failed: Connection to guacd timed out.
Jan 29 19:08:35 bc guacd[1244]: Creating new client for protocol "rdp"
Jan 29 19:08:35 bc guacd: guacd[1244]: INFO:#011Creating new client for protocol "rdp"
Jan 29 19:08:35 bc guacd: guacd[1244]: INFO:#011Connection ID is "$7935e599-7f27-4c4e-937c-39888929a590"
Jan 29 19:08:35 bc guacd[1244]: Connection ID is "$7935e599-7f27-4c4e-937c-39888929a590"
Jan 29 19:08:35 bc guacd[9664]: Support for protocol "rdp" is not installed
Jan 29 19:08:35 bc guacd: guacd[9664]: WARNING:#011Support for protocol "rdp" is not installed
Jan 29 19:08:35 bc guacd[1244]: Connection "$7935e599-7f27-4c4e-937c-39888929a590" removed.
Jan 29 19:08:35 bc guacd: guacd[1244]: INFO:#011Connection "$7935e599-7f27-4c4e-937c-39888929a590" removed.
Jan 29 19:08:50 bc server: 19:08:50.877 [http-bio-8080-exec-73] ERROR o.a.g.w.GuacamoleWebSocketTunnelEndpoint - Creation of WebSocket tunnel to guacd failed: Connection to guacd timed out.`

**Comment by docop** [comment]
this above is from a local one i try quick, but here trying to log over a win10 :

Jan 29 19:11:59 bc guacd[1244]: Creating new client for protocol "rdp"
Jan 29 19:11:59 bc guacd: guacd[1244]: INFO:#011Creating new client for protocol "rdp"
Jan 29 19:11:59 bc guacd[1244]: Connection ID is "$f0b82973-b5bd-41dc-80b4-7a06be002e32"
Jan 29 19:11:59 bc guacd: guacd[1244]: INFO:#011Connection ID is "$f0b82973-b5bd-41dc-80b4-7a06be002e32"
Jan 29 19:11:59 bc guacd[9780]: Support for protocol "rdp" is not installed
Jan 29 19:11:59 bc guacd: guacd[9780]: WARNING:#011Support for protocol "rdp" is not installed
Jan 29 19:11:59 bc guacd[1244]: Connection "$f0b82973-b5bd-41dc-80b4-7a06be002e32" removed.
Jan 29 19:11:59 bc guacd: guacd[1244]: INFO:#011Connection "$f0b82973-b5bd-41dc-80b4-7a06be002e32" removed.
Jan 29 19:12:14 bc server: 19:12:14.809 [http-bio-8080-exec-64] ERROR o.a.g.w.GuacamoleWebSocketTunnelEndpoint - Creation of WebSocket tunnel to guacd failed: Connection to guacd timed out.

**Comment by vm1984** [comment]
> 
> 
> You said in your OP that you used stable, not git as the install source:
> 
> > Guacamole Source: Stable/Git? **stable** / letsencrpy option, default all , 1 mariadb
> 
> If using stable (1.0.0), it would have required freerdp1.x which should work. You can confirm RDP installed properly by looking in the install log, there will be an area while compiling Guacamole server that specifies what protocols are available/installed, basically like you mention you saw. It should say yes for RDP.
> 
> However, if you had the problem with installing from git and NOT stable, then this could explain it. Apache must have just updated 1.1.0 from git to use freerdp2 (been waiting for this).
> 
> So 2 things if you used git as source.
> 
>     1. I do plan to update the script to account for changes like this periodically.
> 
>     2. I do **not** provide support for using the git install source, only stable. Git is too unpredictable and not intended for use in production. The option is only in the script for testing purposes, mostly my own to test features before they release and update the script to get ready.
> 
> 
> So please do the following to test and confirm if there is still an issue or not:
> 
>     1. Use the latest version of the script.
> 
>     2. Install CentOS 7.7. It doesn't matter if you do this from a 7.7 ISO/disc or from 7.5 and update, as long as after running `yum update` you reboot the computer AND confirm the version is 7.7 BEFORE running the script.
> 
>     3. Try the following settings first: stable, self signed cert, nginx defaults, mariadb, no 2FA and no custom extension.
> 
>     4. Enter the option to run the script to install.
> 
>     5. For RDP, confirm that RDP works on the destination machine. IE: try to RDP to it from another machine using something like Microsofts RDP client.
> 
>     6. Setup the connection in Guacamole, just the minimum IP (do not use hostname, use local IP) and the proper port, typically 3389.
> 
> 
> IMPORTANT:
> Before attempting to connect via RDP using Guacamole, open a terminal to the server and run the command `tail -f /var/log/messages`. After running this command, keep that terminal open and then try and connect via RDP in Guac. The terminal window will display messages about the connection. See this for details: [#94 (comment)](https://github.com/Zer0CoolX/guacamole-install-rhel/issues/94#issuecomment-577700138)
> 
> If it doesn't work let me know what it says. If it does work and you want Lets Encrypt you can repeat the process (clean install of CentOS) but using the Lets Encrypt option instead of the self signed option (keeping all else the same in step 3). and again verify if it works or not.
> 
> I hope this helps. As mentioned I see RDP working via the script and self signed cert and do not have the means to test Lets Encrypt right now (hoping to be able in a few weeks time). The keys to narrowing this down are checking the install log, checking other logs like /var/log/messages and ensuring the setup of things external to the script (like the requirements on the network for Lets Encrypt to get a cert) are done properly before hand.

Thanks for your reply.
I've selected the installation from git before. Sorry I forgot to mention it earlier.
Currently I'm testing the new version with rdp2.
I've just make another fresh installation according your instructions: centos 7.7 with updates, stable, self signed cert, nginx defaults, mariadb, no 2FA and no custom extension.
It works perfectly.
The script is great, you did lot of work. Thanks for it.

**Comment by Zer0CoolX** [comment]
@docop and @vm1984 First I apologize, I did not realize a 2nd person was involved in this thread until now.

@docop Your log indicated rdp is not installed. Check the installation log that the script generates as I explained above to see if it says yes/no next to the entry for RDP during install. 

Also confirm you are using the stable options and NOT git. You may also use `yum` or another command to verify if freerdp is installed and if so what version.

Follow my directions here: https://github.com/Zer0CoolX/guacamole-install-rhel/issues/96#issuecomment-579810801 and see if that works for you and if it does then try again with Lets Encrypt (after fresh CentOS install).

@vm1984 As mentioned I dont support install from git and the current situation is a good reason why. Guac 1.1.0 on Github seems to have recently been updated to be able to use freerdp 2.x which is great news as CentOS/RHEL 8 and CentOS/RHEL 7.7 and higher both normally install freerdp 2.x.

The stable version (and early git versions) of Guac can only use freerdp 1.x. Thus my script checks RHEL/CentOS version. 7.6 and under and just does `yum install freerdp` to get freerdp 1.x. If the version is 7.7 or a higher 7.x version, then I forcibly install freerdp 1.x and prevent it from being updated to 2.x via yum update in the future.

However, I do not within that logic also check if the source is git and then install freerdp 2.x (yet). So with the latest revision to Guac from git, freerdp 1.x gets installed, not 2.x and thus it does work.

Now, heres the twist...

Yesterday I tested the script with git as source and stripped out forcing freerdp 1.x, thus installing freerdp 2.x. Script ran, no errors, freerdp 2.x installed, rdp supported. I setup a connection and it wont connect.

There is apparently a quirk/bug between Guac and freerdp 2.x ([seen here](https://issues.apache.org/jira/browse/GUACAMOLE-927)) that prevents the connection from working due to not being able to write to the users "home" folder. It is unclear to me if this is fixed yet and/or pushed to Guacamole on git yet.

Long story short, complications like this happen frequently with using git as source thus why I only support stable as source. I could waste hours or even days trying to fix an issue when using git as the source only to find they change it again to resolve the matter.

**Comment by Zer0CoolX** [comment]
As Guacamole 1.1.0 was released and the script has been updated to work with it, I am going to close this issue for now.

Anyone having issues with this script and RDP please try the latest version of the script + CentOS/RHEL 7.7 (or higher 7.x version). This will install and use freerdp 2.x for RDP which I have tested and found to be working. Be aware you may need to use additional settings in the connection within Guacamole (for example ignore server certificate, encryption type, etc.) even if you did not have to before hand.
